### DISCUSSION TITLE
Sample Code for React Native Apps

### DISCUSSION THREAD

**Discussion by Rummy99** [CoderLine/alphaTab#1740] [Q&A] [answered]
Hi, Is there a sample program for react native app? The samples from CodeSandbox and AlphaTab sites are not compiling.

**Comment by Danielku15** [comment, marked as answer]
Currently there is no React Native sample app and I haven't tried it out yet. 

# alphaTab with React Native using JavaScript

React Native should have general support for any JavaScript Library on the "backend" side of things. That means generally the ["low level APIs"](https://www.alphatab.net/docs/guides/lowlevel-apis) should be available. 

Building a full app requires proper UI integration where currently no platform implementation exists. The platform specific UI integrations take care of following bits:

* Building the bridge between the GUI engine and alphaTab (mapping events like resizing, handling user inputs and layouting the native UI controls displaying the rendered raster graphics)
* Building the bridge between the platform audio and alphaSynth (buffering the generated audio and sending it to the audio output device)
* Ensuring background workers / threading integration (via web workers, or threads)

The platform implementations live in:
* https://github.com/CoderLine/alphaTab/tree/develop/src/platform/javascript
* https://github.com/CoderLine/alphaTab/tree/develop/src.csharp/AlphaTab/Platform
* https://github.com/CoderLine/alphaTab/tree/develop/src.kotlin/alphaTab/android/src/main/java/alphaTab/platform

React native is still handling the UI in the iOS/Android native UI world and therefore standard web components are not working out of the box.

# alphaTab with React native using Android Control

alphaTab also has a native Android control (view) which could be integrated via mechanisms like this:

https://reactnative.dev/docs/0.72/native-components-android

**Reply by Rummy99** [reply]
Thank you for the details.
